src/mono/mono/metadata/class-init.c

```diff
@@ -695,10 +695,7 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 		return NULL;
 	}
 
-	if (klass->image->assembly_name && !strcmp (klass->image->assembly_name, "Mono.Simd") && !strcmp (nspace, "Mono.Simd")) {
-		if (!strncmp (name, "Vector", 6))
-			klass->simd_type = !strcmp (name + 6, "2d") || !strcmp (name + 6, "2ul") || !strcmp (name + 6, "2l") || !strcmp (name + 6, "4f") || !strcmp (name + 6, "4ui") || !strcmp (name + 6, "4i") || !strcmp (name + 6, "8s") || !strcmp (name + 6, "8us") || !strcmp (name + 6, "16b") || !strcmp (name + 6, "16sb");
-	} else if (klass->image->assembly_name && !strcmp (klass->image->assembly_name, "System.Numerics") && !strcmp (nspace, "System.Numerics")) {
+	if (mono_is_corlib_image (klass->image) && !strcmp (nspace, "System.Numerics")) {
```

Comment:
this is actually a bugfix since System.Numerics.Vector4 is in System.Private.CoreLib.dll since https://github.com/dotnet/runtime/commit/4ed5eea1a25e075942f53fd261b29792c7fcc67f instead of System.Numerics.dll

Reply:
this actually uncovered a bug, so I disabled it for now: https://github.com/dotnet/runtime/issues/67762